MainRepository/Program.cs

```diff
@@ -1,7 +1,45 @@
-﻿internal class Program
-{
-    private static void Main(string[] args)
-    {
-        Console.WriteLine("Hello, World!");
-    }
-}
\ No newline at end of file
+﻿// Implementation of string is valid file path or not into tuple format.
+using static State;
+var strs = new List<string> { @"C:\PROGRAMS\DATA\README.TXT", @"C:\PROGRAMS\README.TXT", @"C:\PROGRAMS\DATA\DIR\README.TXT", @"C:\README.TXT", "..", "", "C:", "C:\\PROGRAMS\\README..TXT" };
+foreach (var str in strs) {
+   if (ParseFile (str)) {
+      Console.WriteLine ($"{str} is a Valid filepath.");
+      var res = Result (str);
+      Console.WriteLine ($"{res.dir}, {res.folder}, {res.filename}, {res.ext}");
+   } else { Console.WriteLine ($"{str} is Not a Valid filepath."); }
+}
+
+// Convert valid filepath into tuple format
+(string dir, string folder, string filename, string ext) Result (string input) {
+   string[] splits = input.Split ('\\'); // C: programs data readme.txt
+   string drive = splits[0];
+   string folders = "";
+   for (int i = 1; i < splits.Length - 1; i++) folders += splits[i] + "\\";
+   splits = splits[splits.Length - 1].Split ('.'); // readme txt
+   string filename = splits[0];
+   string extension = splits[1];
+   return (drive, folders, filename, extension);
+}
+
+// Check string is valid filepath or not
+bool ParseFile (string input) {

```

Comment:
Name of the method not so relevant.
```suggestion
bool IsValidFilePath (string input) {

```
```suggestion
bool ParseFile (string input) {

```
If this has to be ParseFile you could return the drive, folder etc in this method.